quickstart/201-aks-log-analytics/aks.tf

```diff
@@ -1,27 +1,36 @@
+resource "azurerm_user_assigned_identity" "aks_identity" {
+  location            = azurerm_resource_group.default.location
+  name                = "${var.name}-aks-identity"
+  resource_group_name = azurerm_resource_group.default.name
+}
+
+resource "azurerm_role_assignment" "default" {
+  principal_id = azurerm_user_assigned_identity.aks_identity.principal_id
+  scope        = azurerm_resource_group.default.id
+  role_definition_name = "Network Contributor"
+}
+
 resource "azurerm_kubernetes_cluster" "default" {
   name                = "${var.name}-aks"
-  location            = "${azurerm_resource_group.default.location}"
-  resource_group_name = "${azurerm_resource_group.default.name}"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
   dns_prefix          = "${var.dns_prefix}-${var.name}-aks-${var.environment}"
-  depends_on          = ["azurerm_role_assignment.default"]
+  role_based_access_control_enabled = true
+  depends_on          = [azurerm_role_assignment.default]
 
-  agent_pool_profile {
+  default_node_pool {
     name            = "default"
-    count           = "${var.node_count}"
-    vm_size         = "${var.node_type}"
-    os_type         = "Linux"
+    node_count           = var.node_count
+    vm_size         = var.node_type
     os_disk_size_gb = 30
   }
 
-  service_principal {
-    client_id     = "${azuread_application.default.application_id}"
-    client_secret = "${azuread_service_principal_password.default.value}"
-  }
+identity {
+  type = "UserAssigned"
+  identity_ids = [azurerm_user_assigned_identity.aks_identity.id]
+}
 
-  addon_profile {
-    oms_agent {
-      enabled                    = true
-      log_analytics_workspace_id = "${azurerm_log_analytics_workspace.default.id}"
-    }
+  microsoft_defender {
```

Comment:
There're two `log_analytics_workspace_id` in `azurerm_kubernetes_cluster` resource, I think we should use [`oms_agent` block](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#oms_agent) with `log_analytics_workspace_id` here.

Reply:
Right, changed to `oms_agent`.